src/core/array.d

```diff
@@ -0,0 +1,161 @@
+module core.array;
+
+import core.internal.traits : Unqual, CommonType;
+
+debug import std.stdio;
+
+// @nogc: https://issues.dlang.org/show_bug.cgi?id=18439
+nothrow @safe pure:
+/++
+    Returns a static array constructed from `a`. The type of elements can be
+    specified implicitly (`int[2] a = staticArray(1,2);`) or explicitly
+    (`float[2] a = staticArray!float(1,2)`).
+
+    The result is an rvalue, therefore uses like staticArray(1, 2, 3).find(x) may be inefficient.
```

Comment:
Maybe providing some links to what an rvalue is or explaining the inefficiency a bit better. Just from the stand point of someone that isn't familiar with what rvalues are and why this might be inefficient. Using `find(staticArray(...), x)` might better illustrate that point as well.

Reply:
> Maybe providing some links to what an rvalue is 

i think that's out of scope; other functions mentioning rvalue don't explain either

> explaining the inefficiency a bit better. 
done

> find(staticArray(...), x) 

changed it to fun(staticArray(...))